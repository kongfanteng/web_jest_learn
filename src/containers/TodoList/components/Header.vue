<template>
  <div class="header">
    <div class="header-content">
      TodoList
      <input
        data-test="input"
        type="text"
        v-model="inputValue"
        @keyup.enter="addTodoItem"
        class="header-input"
        placeholder="输入待办事项"
      />
    </div>
  </div>
</template>

<script>
export default {
  name: "TodoListHeader",
  data() {
    return {
      inputValue: "",
    };
  },
  methods: {
    addTodoItem() {
      if (this.inputValue) {
        this.$emit("add", this.inputValue);
        this.inputValue = "";
      }
    },
  },
};
</script>
<style scoped lang="stylus">
.header
  line-height 60px
  background #334
.header-content
  width 600px
  margin 0 auto
  color #fff
  font-size 24px
.header-input
  float right
  width 360px
  margin-top 16px
  line-height 24px
  outline none
  color #333
  text-indent 8px
</style>
