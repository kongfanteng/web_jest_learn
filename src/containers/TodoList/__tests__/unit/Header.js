import { shallowMount } from "@vue/test-utils";
import Header from "../../components/Header.vue";
import { findTestWrapper } from "@/utils/testUtils";
describe("Header 组件", () => {
  it("样式发生改变，需做提示", () => {
    const wrapper = shallowMount(Header);
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("输入框存在", () => {
    const wrapper = shallowMount(Header);
    const input = findTestWrapper(wrapper, "input").at(0);
    expect(input.exists()).toBe(true);
  });
  it("输入框初始内容为空", () => {
    const wrapper = shallowMount(Header);
    const inputValue = wrapper.vm.$data.inputValue;
    expect(inputValue).toBe("");
  });
  it("输入框值发生变化时，数据跟随变化", () => {
    const wrapper = shallowMount(Header);
    const input = findTestWrapper(wrapper, "input").at(0);
    input.setValue("hello");
    const inputValue = wrapper.vm.$data.inputValue;
    expect(inputValue).toBe("hello");
  });
  it("输入框回车时，无内容，无反应", () => {
    const wrapper = shallowMount(Header);
    const input = findTestWrapper(wrapper, "input").at(0);
    input.setValue("");
    input.trigger("keyup.enter");
    expect(wrapper.emitted().add).toBeFalsy();
  });
  it("输入框回车时，有内容时，向外触发事件，同时清空 inputValue", () => {
    const wrapper = shallowMount(Header);
    const input = findTestWrapper(wrapper, "input").at(0);
    input.setValue("hello");
    input.trigger("keyup.enter");
    expect(wrapper.emitted().add).toBeTruthy();
    expect(wrapper.vm.$data.inputValue).toBe("");
  });
});
