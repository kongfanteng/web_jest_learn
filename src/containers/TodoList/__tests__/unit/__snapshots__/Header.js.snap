// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header 样式发生改变，进行提示 1`] = `
<div class="header">
  <div class="header-content"> TodoList <input data-test="input" type="text" class="header-input" placeholder="输入待办事项"></div>
</div>
`;
